src/engine/Operation.cpp

```diff
@@ -72,69 +69,41 @@ shared_ptr<const ResultTable> Operation::getResult(bool isRoot) {
-      abort(newResult, true);
-      // Rethrow as QUERY_ABORTED allowing us to print the Operation
-      // only at innermost failure of a recursive call
-      throw ad_semsearch::AbortException(e);
-    } catch (...) {
-      // We are in the innermost level of the exception, so print
-      abort(newResult, true);
-      LOG(ERROR) << "WEIRD_EXCEPTION not inheriting from std::exception"
-                 << endl;
-      // Rethrow as QUERY_ABORTED allowing us to print the Operation
-      // only at innermost failure of a recursive call
-      throw ad_semsearch::AbortException("WEIRD_EXCEPTION");
-    }
-    timer.stop();
-    _runtimeInfo.setRows(newResult->_resTable->size());
-    _runtimeInfo.setCols(getResultWidth());
-    _runtimeInfo.setDescriptor(getDescriptor());
-    _runtimeInfo.setColumnNames(getVariableColumns());
+  try {
+    auto computeLambda = [&, this] {
+      CacheValue val(getExecutionContext()->getAllocator());
+      computeResult(val._resTable.get());
+      return val;
+    };
 
-    _runtimeInfo.setTime(timer.msecs());
-    _runtimeInfo.setWasCached(false);
-    // cache the runtime information for the execution as well
-    newResult->_runtimeInfo = _runtimeInfo;
-    // Only now we can let other threads access the result
-    // and runtime information
-    newResult->_resTable->finish();
-    return newResult->_resTable;
-  }
+    auto result = (pinResult) ? cache.computeOncePinned(cacheKey, computeLambda)
+                              : cache.computeOnce(cacheKey, computeLambda);
 
-  existingResult->_resTable->awaitFinished();
-  if (existingResult->_resTable->status() == ResultTable::ABORTED) {
-    LOG(ERROR) << "Operation aborted while awaiting result" << endl;
-    AD_THROW(ad_semsearch::Exception::BAD_QUERY,
-             "Operation aborted while awaiting result");
+    timer.stop();
+    createRuntimeInformation(result, timer.msecs());
+    return result._resultPointer->_resTable;
+  } catch (const ad_semsearch::AbortException& e) {
+    // A child Operation was aborted, do not print the information again.
+    throw;
+  } catch (const ad_utility::AbortedByOtherThreadException& e) {
+    LOG(ERROR) << "Aborted operation was found in the cache:" << std::endl;
+    LOG(ERROR) << asString();
+    throw ad_semsearch::AbortException(e);
+  } catch (const std::exception& e) {
+    // We are in the innermost level of the exception, so print
+    LOG(ERROR) << "Aborted Operation:" << endl;
+    LOG(ERROR) << asString() << endl;
+    LOG(ERROR) << e.what() << endl;
+    // Rethrow as QUERY_ABORTED allowing us to print the Operation
+    // only at innermost failure of a recursive call
+    throw ad_semsearch::AbortException(e);
+  } catch (...) {
+    // We are in the innermost level of the exception, so print
+    LOG(ERROR) << "Aborted Operation:" << endl;
+    LOG(ERROR) << asString() << endl;
+    LOG(ERROR) << "WEIRD_EXCEPTION not inheriting from std::exception" << endl;
```

Comment:
The answer to this seems to be: not really. And since it is at the end of a well structured function, 
it doesn't bother me too much. We could extract the common strings, but in this case it would not really make the code more readable.